### PR TITLE
Use new scheduler when doing a schema update

### PR DESCRIPTION
**Pull request by benmoriceau**
## What

Use the new scheduler upon updating the schema of a connection.

### CODE REVIEW

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java [hunk 1]
```diff
@@ -357,7 +357,33 @@ public ManualSyncSubmissionResult resetConnection(final UUID connectionId) {
...
+    log.info("end of reset submission");
+
+    final long jobId = connectionManagerWorkflow.getJobInformation().getJobId();
+
+    return new ManualSyncSubmissionResult(
+        Optional.empty(),
+        Optional.of(jobId));
+  }
+
+  public ManualSyncSubmissionResult synchronousResetConnection(UUID connectionId) {
```
Comment (jrhizor) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java:369]: I think this needs a doc string. It isn't intuitive from reading that the condition it's waiting for is that the connection manager switches to a new job after completing the reset flow.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java [hunk 2]
```diff
@@ -357,7 +357,33 @@ public ManualSyncSubmissionResult resetConnection(final UUID connectionId) {
...
+    resetConnection(connectionId);
```
Comment (jrhizor) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java:370]: Should this fail if a `ManualSyncSubmissionResult` is returned without a workflow running?

File: airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java [hunk 3]
```diff
@@ -535,6 +541,46 @@ void testUpdateConnectionWithUpdatedSchema() throws JsonValidationException, Con
...
+            .status(expected.getStatus())
+            .schedule(expected.getSchedule()));
+    when(featureFlags.usesNewScheduler()).thenReturn(true);
+
+    final WebBackendConnectionRead connectionRead = wbHandler.webBackendUpdateConnection(updateBody);
+
+    assertEquals(expectedWithNewSchema.getSyncCatalog(), connectionRead.getSyncCatalog());
+
+    final ConnectionIdRequestBody connectionId = new ConnectionIdRequestBody().connectionId(connectionRead.getConnectionId());
+    verify(schedulerHandler, times(0)).resetConnection(connectionId);
```
Comment (jrhizor) [inline airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java:578]: For these operations it's very important that the calls aren't just made, but that they also happen in the correct order. Can we verify that here with something like https://stackoverflow.com/a/40411151/175798 ?

**Review by jrhizor — APPROVED** [review]
Only minor comments. 

I think we should also have a followup ticket for handling this completely async. I think in the future we should want a queueing mechanism on the connection updater workflow for just the reset, a reset + sync, etc. that works as expected even if a sync is currently running. We'd have to figure out what the user would have to see for this too.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java (hunk 2 above)
Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java:370]: Done

File: airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java (hunk 3 above)
Comment (benmoriceau) [reply airbyte-server/src/test/java/io/airbyte/server/handlers/WebBackendConnectionsHandlerTest.java:578]: Done

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java (hunk 1 above)
Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java:375]: Done

**Comment by benmoriceau** [comment]
Use the new scheduler upon updating the schema of a connection.

**Pull request merged**